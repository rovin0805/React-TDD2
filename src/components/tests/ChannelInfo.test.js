import { render, screen, waitFor } from '@testing-library/react';
import { Route } from 'react-router-dom';
import { withAllContexts, withRouter } from '../../tests/utils';
import ChannelInfo from '../ChannelInfo';

describe('ChannelInfo', () => {
  const fakeYoutube = {
    channelImageURL: jest.fn(),
  };

  const name = 'channelName';

  function renderChannelInfoWithCallback(callback) {
    fakeYoutube.channelImageURL.mockImplementation(callback);
    return render(
      withAllContexts(
        withRouter(
          <Route path='/' element={<ChannelInfo id='id' name={name} />} />
        ),
        fakeYoutube
      )
    );
  }

  afterEach(() => {
    fakeYoutube.channelImageURL.mockReset();
  });

  it('renders correctly', async () => {
    const { asFragment } = renderChannelInfoWithCallback(() => 'url');

    await waitFor(() => {
      screen.getByText(name);
    });

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders without URL', () => {
    renderChannelInfoWithCallback(() => {
      throw new Error('error');
    });

    expect(screen.queryByRole('img')).toBeNull();
  });

  it('renders with URL', async () => {
    renderChannelInfoWithCallback(() => 'url');

    const img = await screen.findByRole('img');
    expect(img).toBeInTheDocument();
  });
});
