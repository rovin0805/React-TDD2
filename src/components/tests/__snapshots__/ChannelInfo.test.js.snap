// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelInfo renders correctly 1`] = `
<DocumentFragment>
  <div
    class="flex my-4 mb-8 items-center"
  >
    <p
      class="text-lg font-medium ml-2"
    >
      channelName
    </p>
  </div>
</DocumentFragment>
`;
